include/cxx.h

```diff
@@ -235,15 +235,27 @@ class Vec final {
 
   class const_iterator {
   public:
+    using difference_type = ptrdiff_t;
     using value_type = typename std::add_const<T>::type;
+    using pointer = typename std::add_pointer<
+        typename std::add_const<T>::type>::type;
     using reference = typename std::add_lvalue_reference<
         typename std::add_const<T>::type>::type;
+    using iterator_category = std::forward_iterator_tag;
 
+    const_iterator(): pos(nullptr), stride(0) { }
+    const_iterator(const void *pos, size_t stride): pos(pos), stride(stride) { }
```

Comment:
I would prefer not to offer this constructor. Only `friend class Vec` should be creating non-default iterators.

Reply:
Makes sense.  It made more sense with some code I had before, but now it is competely unnecessary.  Removed in 72ceca3.